### ISSUE TITLE
Attempting to update more than 10 WordPress posts at once throws exception.

### ISSUE THREAD

**Issue by GlenAtPantheon** [pantheon-systems/WordPress#24]
Error message: Warning: Attempted to clear more than 10 urls in pantheon.php on line 375 causes AJAX request failure

Edge cache clear with more than 10 urls invoked on AJAX call leading to plain text error in response which prevents JSON being correctly interpreted

The solution here is to switch from the BAN api to the PURGE api.

**Comment by sukottokun** [comment]
@GlenAtPantheon Hey, just catching up here, is this something the customer can do or an eng solution to the user's issue.

**Comment by ttrowell** [comment]
Team,  I have this issue happening with a client that has only 1 ajax call.  Is there a reason the Error message would be triggered under the 10 url threshold?

**Comment by nstielau** [comment]
What are the parameters to the 1 ajax call?  It’s not about the number of
calls, but the number of URIs passed in.

On Wed, Oct 14, 2015 at 9:14 AM, Tara Rowell <notifications@github.com>
wrote:

> Team, I have this issue happening with a client that has only 1 ajax call.
> Is there a reason the Error message would be triggered under the 10 url
> threshold?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pantheon-systems/WordPress/issues/24#issuecomment-148045969>
> .
>



-- 
Nick Stielau | p 415-515-2120
Director of Engineering | Pantheon

**Comment by oppodeldoc** [comment]
Hi Nick, thanks for looking into this issue. 

We don't have multiple URI's being passed into the call, and in fact, the Ajax request is being completed successfully on the server and returning a success message. The problem is that this notification is being appended as a string to the response and is malforming the JSON. 

A sample Ajax call would be a post to this address:
**AJAXURL**id=174&action=update_post&security=92e85b66f2&post_type=_rcylst_guide&_rcylst_what_to_do=12&field_54d2a6cc650d0=&field_54e68e6538ff8=false

I'm honestly not clear what would trigger this, can you give an example as to what would cause the 10 URL limit warning to show up? Right now it is intermittent for us, but when it happens it breaks the front end of our application.

**Comment by nstielau** [comment]
I believe the 10 URLs include images, thumbnails, etc related to the post/page itself.  You can try preventing the `E_USER_WARNING` error from being printed with [`error_reporting`](http://php.net/manual/en/function.error-reporting.php) functionl

**Comment by oppodeldoc** [comment]
Ok, thanks for the information. Any best-practices for where to change error reporting? Should I set it in the wp-config file?

I'm a little concerned about just suppressing this warning without knowing exactly why it's showing up in the first place, and also, it looks to be intermittent.  I'm seeing nothing in the logs on any of my installs, and it's technically working on the server side, is there something wrong with the Params that I posted?  

Thanks again for your help!

**Comment by populist** [comment]
Here is the documentation on error reporting (http://php.net/manual/en/function.error-reporting.php) which can be used to suppress this message on AJAX calls until the switch from the BAN api to the PURGE api is complete.

**Comment by riteshtailordiaspark** [comment]
I am also getting same error although i am using edgecast webservice API to purge URL but that invoked only after saving post.
Any help would be much appreciable.

Thanks

**Comment by mattking5000** [comment]
I was debugging a plugin that worked on LIVE but not on DEV and traced it down to this as well...seems like this warning is somewhat meaningless, yet it breaks some plugins that rely on ajax. Is this still being fixed?

**Comment by scottbasgaard** [comment]
Experiencing this issue as well, any updates on if the issue is on implementation or Pantheon's side?

**Comment by jhabdas** [comment]
Bump

**Comment by joshkoenig** [comment]
This should be getting fixed as part of an upcoming feature release. We are now tracking this internally as a bug report. Apologies for the long-lived issue.

**Comment by joshkoenig** [comment]
@danielbachhuber can we also take a look at this (possibly updating our base cache-busting API call) as part of the batching work for the Advanced Page Cache

**Comment by danielbachhuber** [comment]
@joshkoenig I'm pretty sure this was resolved with the upstream release in December. Specifically, we removed the call to `enqueue_urls()` with flushing the post cache: https://github.com/pantheon-systems/WordPress/pull/88/files#diff-57765acdd6d4f9f63197e76e9e76f44fL302 At this point, the upstream only handles calls to flush the entire site.

@riteshtailordiaspark @mattking5000 @scottbasgaard Have you seen this issue on any of your sites in the last six months?

With this being said, it seems like we have a similar issue in Pantheon Advanced Page Cache (https://github.com/pantheon-systems/pantheon-advanced-page-cache/issues/95, https://github.com/pantheon-systems/pantheon-advanced-page-cache/issues/86) that we'll hopefully be able to address by batching the calls.
